### PR TITLE
[SDK-2597] Update README.md and UPGRADE.md for 8.x

### PR DESCRIPTION
**Pull request by evansims**
This PR updates the `UPGRADE.md` (previously `MIGRATE-v5-TO-v7.md`) and `README.md` files with an overview of changes from 7.x to 8.x, and guidance on how developers can approach upgrading their applications.

The PR also includes some minor unit test bug fixes, docblock grammar improvements, and other minor tweaks.

The relevant files of focus for review are:
[README.md](https://github.com/auth0/auth0-PHP/blob/3851f9a46cfe0fe80ea514d968d0473928e5ebed/README.md)
[UPGRADE.md](https://github.com/auth0/auth0-PHP/blob/3851f9a46cfe0fe80ea514d968d0473928e5ebed/UPGRADE.md)

### CODE REVIEW

File: src/Auth0.php [hunk 1]
```diff
@@ -108,6 +109,24 @@ public function login(
         header('Location: ' . $this->authentication()->getLoginLink($params));
     }
 
+    /**
```
Comment (evansims) [inline src/Auth0.php:112]: Unrelated to this PR, but this suggestion came through via the community forums and I thought it was a good one, so I'm sneaking it in here.

**Review by stevehobbsdev — CHANGES_REQUESTED** [review]
Hey @evansims, nice work here, you've obviously put a lot into it. I just left some mostly minor stuff for you.

I've had a read through the upgrade doc and the overriding feeling I have about it is that there isn't really one list or round up of the things that would actually break me if I was to uprgade to the latest version. Generally, it's there, but I really have to dig into the detail to figure out if I'm affected or not, amongst wording like "You can now use the.." and "SDK v8.0 offers a cleaner approach". What I was expecting to see is a heading that is literally "BREAKING CHANGES" (although not necessarily in caps 😄 ) and then a bullet-list of what I should be concerned about as a developer. Then all the awesome detail about the changes can come after. Is that possible here?

Otherwise, nice work 👍🏻

File: README.md [hunk 2]
```diff
@@ -1,31 +1,30 @@
-<p align="center"><a href="https://auth0.com" target="_blank"><img src=".github/logo.svg?sanitize=true&raw=true" width="300"></a></p>
+# Auth0 PHP SDK
 
-<p align="center">
-<a href="https://circleci.com/gh/auth0/auth0-PHP"><img src="https://img.shields.io/circleci/project/github/auth0/auth0-PHP/master.svg" alt="Build Status"></a>
-<a href="https://packagist.org/packages/auth0/auth0-PHP"><img src="https://img.shields.io/packagist/dt/auth0/auth0-PHP" alt="Total Downloads"></a>
-<a href="https://packagist.org/packages/auth0/auth0-PHP"><img src="https://img.shields.io/packagist/v/auth0/auth0-PHP?label=stable" alt="Latest Stable Version"></a>
-<a href="https://packagist.org/packages/auth0/auth0-PHP"><img src="https://img.shields.io/packagist/php-v/auth0/auth0-php" alt="PHP Support"></a>
```
[the remark this answers is not in the document]
Comment (evansims) [reply README.md:7]: That's fair, I'll re-add some of these

File: README.md [hunk 3]
```diff
@@ -35,232 +34,484 @@ This is [one of many libraries we offer](https://auth0.com/docs/libraries) suppo
 
 ## Requirements
 
-- PHP 7.3+ / 8.0+
+- PHP [7.4](https://www.php.net/ChangeLog-7.php) or [8.0](https://www.php.net/ChangeLog-8.php)
 - [Composer](https://getcomposer.org/)
+- A PSR-17 HTTP factory library. (↗ [Find libraries](https://packagist.org/search/?query=PSR-17&type=library&tags=psr%2017))
+- A PSR-18 HTTP client library. (↗ [Find libraries](https://packagist.org/search/?query=PSR-18&type=library&tags=psr%2018))
+- A PSR-16 caching library is strongly recommended for performance reasons. (↗ [Find libraries](https://packagist.org/search/?query=PSR-16&type=library&tags=psr%2016))
```
[the remark this answers is not in the document]
Comment (evansims) [reply README.md:41]: Good call, I'll link these references to them as well

**Comment by evansims** [comment]
Hey @stevehobbsdev, thanks for the awesome feedback! I've tried to incorporate everything you suggested now. I've surfaced the more potentially breaking changes (depending on how the developer may be using the SDK, of course) in the UGPRADE.md file. See what you think of this approach 🤔

File: README.md [hunk 4]
```diff
@@ -35,232 +36,474 @@ This is [one of many libraries we offer](https://auth0.com/docs/libraries) suppo
...
 ```
 
-**Note:** _In a production application you should never hardcode these values. Consider using environment variables to store and pass these values to your application, as suggested in our [documentation](https://auth0.com/docs/libraries/auth0-php#getting-started)._
+> ⚠️ You should **never** hard-code tokens or other sensitive configuration data in a real-world application. Consider using environment variables to store and pass these values to your application.
+
+This method is discouraged because you lose out on type hinting, but is useful in PHP 7.4 where named arguments are not supported.
+
+The following options are available for your configuration:
+
+| Option Name           | Allowed Types                     | Required | Default                          | Description                                                                                                                                                                                                                             |
```
Comment (Nyholm) [inline README.md:165]: This table is appreciated. Thank you

**Review by Nyholm — COMMENTED** [review]
There could be a good idea to keep all the old upgrade documents. 
Ie, Upgrade-5-to-7.md and Upgrade-7-to-8.md.

**Pull request merged**